include/SDL3/SDL_video_capture.h

```diff
@@ -0,0 +1,367 @@
+ * \returns number of framesizes or a negative error code on failure; call
+ *          SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_GetVideoCaptureFrameSize
+ * \sa SDL_SetVideoCaptureSpec
+ */
+extern DECLSPEC int SDLCALL SDL_GetNumVideoCaptureFrameSizes(SDL_VideoCaptureDevice *device, Uint32 format);
+
+
+/**
+ * Get video capture status
+ *
+ * \param device opened video capture device
+ * \returns 0 on success or a negative error code on failure; call
+ *          SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_VideoCaptureStatus
+ */
+extern DECLSPEC SDL_VideoCaptureStatus SDLCALL SDL_GetVideoCaptureStatus(SDL_VideoCaptureDevice *device);
+
+/**
+ * Start video capture
+ *
+ * \param device opened video capture device
+ * \returns 0 on success or a negative error code on failure; call
+ *          SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_StopVideoCapture
+ */
+extern DECLSPEC int SDLCALL SDL_StartVideoCapture(SDL_VideoCaptureDevice *device);
+
+/**
+ * Acquire a frame.
+ * The frame is a memory pointer to the image data, whose size and format
+ * are given by the the obtained spec.
+ *
+ * Non blocking API. If there is a frame available, frame->num_planes is non 0.
+ * If frame->num_planes is 0 and returned code is 0, there is no frame at that time.
+ *
+ * After used, the frame should be released with SDL_ReleaseVideoCaptureFrame
+ *
+ * \param device opened video capture device
+ * \param frame pointer to get the frame
+ * \returns 0 on success or a negative error code on failure; call
+ *          SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_ReleaseVideoCaptureFrame
+ */
+extern DECLSPEC int SDLCALL SDL_AcquireVideoCaptureFrame(SDL_VideoCaptureDevice *device, SDL_VideoCaptureFrame *frame);
+
+/**
+ * Release a frame. Let the back-end re-use the internal buffer for video capture.
+ *
+ * \param device opened video capture device
+ * \param frame frame pointer.
+ * \returns 0 on success or a negative error code on failure; call
+ *          SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_AcquireVideoCaptureFrame
+ */
+extern DECLSPEC int SDLCALL SDL_ReleaseVideoCaptureFrame(SDL_VideoCaptureDevice *device, SDL_VideoCaptureFrame *frame);
+
+/**
+ * Stop Video Capture
+ *
+ * \param device opened video capture device
+ * \returns 0 on success or a negative error code on failure; call
+ *          SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_StartVideoCapture
+ */
+extern DECLSPEC int SDLCALL SDL_StopVideoCapture(SDL_VideoCaptureDevice *device);
+
+/**
+ * Use this function to shut down video_capture processing and close the video_capture device.
```

Comment:
What happens if a `SDL_VideoCaptureFrame` is not released prior to closing the capture device?
Does `SDL_ReleaseVideoCaptureFrame` after `SDL_CloseVideoCapture` still work (and not leak)?

Reply:
yes, good point...
currently SDL_ReleaseVideoCaptureFrame need the "device", so a releaseFrame won't work.

maybe for some driver, it's not possible to use the frame after the device is closed.
for some other maybe it would be possible.

We could for the release all frame in the CloseDevice (or StopDevice?)
or prevent a CloseDevice if some frame a not released ? 
also. frame that are recorded but not acquired would also need to be release :)
-> so probably, best is to release all frame (acquirer or not) in CloseDevice

Reply:
(btw CloseDevice invalidates the 'device', so SDL_ReleaseVideoCaptureFrame  would do invalid memory read/write)

Reply:
done here: https://github.com/libsdl-org/SDL/pull/5477/commits/6b3ec97448b1f1818cdba9e1ac259d2d31eaa852

Reply:
ok, I am wrong. here. 
when a frame is acquired, it has to be release before closing the device
- on android, it's an AImage, so we could release the image anyway
- but on linux, we need the file descriptor of the device

Reply:
(but we still need to release the non acquired frames!!) so the commit is ok

Reply:
I've added a comment in the public header, for ReleaseFrame()